vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/HttpRequest.java

```diff
@@ -37,8 +40,8 @@ public byte[] body() {
         return body;
     }
 
-    public Duration timeout() {
-        return timeout;
+    public Duration timeLeft() {
+        return Duration.ofMillis(deadlineMillis - clock.getAsLong());
```

Comment:
Consider using nano time to more accurately measure elapsed time (that is not affected by wall clock adjustments etc).